packages/dds/map/src/mapKernel.ts

```diff
@@ -629,10 +926,28 @@ export class MapKernel {
 						pendingMessageIds[0] === localOpMetadata.pendingMessageId,
 					0x2fa /* Unexpected pending message received */,
 				);
-				pendingMessageIds.shift();
+				const pendingMessageId = pendingMessageIds.shift();
 				if (pendingMessageIds.length === 0) {
 					this.pendingKeys.delete(op.key);
 				}
+
+				if (op.type === "set") {
+					this.ackPendingSetOp(op, pendingMessageId);
+				} else if (op.type === "delete") {
+					// Adjust the keys order if it is already ack'd
```

Comment:
This comment doesn't look like it applies here.

Reply:
Added more detailed comments